packages/cardano-services/test/StakePoolSeach/DbSyncStakePoolSearchProvider/StakePoolSearchBuilder.test.ts

```diff
@@ -0,0 +1,149 @@
+import { Cardano, StakePoolQueryOptions } from '@cardano-sdk/core';
+import { Pool } from 'pg';
+import { StakePoolSearchBuilder } from '../../../src';
+
+describe('StakePoolSearchBuilder', () => {
+  let dbConnection: Pool;
+  let builder: StakePoolSearchBuilder;
+
+  beforeAll(async () => {
+    dbConnection = new Pool({ connectionString: process.env.DB_CONNECTION_STRING });
+    builder = new StakePoolSearchBuilder(dbConnection);
+  });
+
+  afterAll(async () => {
+    await dbConnection.end();
+  });
+
+  const filters: StakePoolQueryOptions['filters'] = {
+    _condition: 'or',
+    identifier: {
+      _condition: 'and',
+      values: [
+        { name: 'CL' },
+        { ticker: 'CLIO' },
+        { id: Cardano.PoolId('pool1jcwn98a6rqr7a7yakanm5sz6asx9gfjsr343mus0tsye23wmg70') }
+      ]
+    },
+    pledgeMet: true,
+    status: Object.values(Cardano.StakePoolStatus)
+  };
+
+  describe('queryRetirements', () => {
+    it('queryRetirements', async () => {
+      const retirements = await builder.queryRetirements([1, 2, 3]);
+      expect(retirements).toMatchSnapshot();
+    });
+  });
+  describe('queryRegistrations', () => {
+    it('queryRegistrations', async () => {
+      const registrations = await builder.queryRegistrations([1, 2, 3]);
+      expect(registrations).toMatchSnapshot();
+    });
+  });
+  describe('queryPoolRewards', () => {
+    it('queryPoolRewards', async () => {
+      const epochRewards = await builder.queryPoolRewards([1, 2, 3]);
+      expect(epochRewards).toMatchSnapshot();
+    });
+  });
+  describe('queryPoolRelays', () => {
+    it('queryPoolRelays', async () => {
+      // todo: change ids
+      const relays = await builder.queryPoolRelays([1, 2, 3]);
+      expect(relays).toEqual([]);
+    });
+  });
+  describe('queryPoolOwners', () => {
+    it('queryPoolOwners', async () => {
+      const owners = await builder.queryPoolOwners([1, 2, 3]);
+      expect(owners).toMatchSnapshot();
+    });
+  });
+  describe('queryPoolMetrics', () => {
+    it('queryPoolMetrics', async () => {
+      const totalAda = '42021479194505231';
+      const metrics = await builder.queryPoolMetrics([1, 2, 3], totalAda);
+      expect(metrics).toMatchSnapshot();
+    });
+  });
+  describe('queryPoolData', () => {
+    it('queryPoolData', async () => {
+      const poolDatas = await builder.queryPoolData([1, 6]);
+      expect(poolDatas).toMatchSnapshot();
+    });
+  });
+  describe('getTotalAmountOfAda', () => {
+    it('getTotalAmountOfAda', async () => {
+      const totalAdaAmount = await builder.getTotalAmountOfAda();
+      expect(totalAdaAmount).toMatchSnapshot();
+    });
+  });
+  describe('getLastEpoch', () => {
+    it('getLastEpoch', async () => {
+      const lastEpoch = await builder.getLastEpoch();
+      expect(lastEpoch).toMatchSnapshot();
+    });
+  });
+  describe('buildPoolsByStatusQuery', () => {
```

Comment:
It would be good to add test for query itself, not only query builder

Reply:
The query itself is being used to build subqueries at `buildOrQuery` and `buildAndQuery`, and then run at queryPoolHashes. So it should be already tested there. Anyway I added a few more validations that includes running the query.